doc/languages-frameworks/python.section.md

```diff
@@ -130,12 +130,12 @@ buildPythonPackage rec {
     hypothesis
   ];
 
-  meta = with lib; {
+  meta = {
     changelog = "https://github.com/pytest-dev/pytest/releases/tag/${version}";
     description = "Framework for writing tests";
     homepage = "https://github.com/pytest-dev/pytest";
-    license = licenses.mit;
-    maintainers = with maintainers; [ domenkozar lovek323 madjar lsix ];
+    license = lib.licenses.mit;
+    maintainers = with lib.maintainers; [ domenkozar lovek323 madjar lsix ];
   };
 }
```

Comment:
I think this is overzealous.

Reply:
Could you please elaborate the reasons?

Reply:
Because it makes meta simpler, and there is no danger of getting things from lib, when they are part of the build function. E.g. `version` will not be pulled from `lib.version`.

Reply:
Thanks now I understand your position.

I mostly agree, technically this is true. However, I saw many PRs where people were removing it from some files, especially NixOS modules. Having it on top of the file is definitely confusing and could potentially make the reasoning a bit harder in some cases.

I personally don't like it much, it's confusing and I never feel the need to use it... Only with lists sometimes, but it's rare.

Now, I don't judge what this PR is doing but I believe that using it with parsimony is fine.

IMHO, this PR is right for me, I don't really find a justification for using it, and therefore it should be removed.

Reply:
>  I found more than 12k+ matches for lib.mdDoc, and more than 50 such appearances on systemd-unit-options.nix, why?

Likely because they were added for all descriptions during the docs migration.

Reply:
> Likely because they were added for all descriptions during the docs migration.

And they continue to be added by novices - I bet a chocolate bar they don't know why `lib.mdDoc` is there since `with lib;` is already there.

Reply:
> Also, as said by eclairevoyant in [#208242 (comment)](https://github.com/NixOS/nixpkgs/issues/208242#issuecomment-1700396671), the documentation encourages that reckless use of `with lib;`.

NB my comment was merely meant to point out that the docs were inconsistent with the tracking issue, and that inconsistency should be rectified, because I'd like to avoid future back-and-forth in PR reviews. 

Despite being a `with`-hater myself, I do not have a strong opinion on `meta = with lib;`, other than avoiding that pattern seems less contentious than using it.

Reply:
@eclairevoyant thanks for expanding and explaining your point.
I didn't  `@`-mark you because I didn't want to bother you with an old reference.

Apologies for not taking that context into account.

On the other hand my reasoning was justifiable: when the documentation includes examples of  `with lib;`, then the documentation encourages its usage.